### ISSUE TITLE
failedSaves contains missed targets even though no save rolled

### ISSUE THREAD

**Issue by tposney** [tposney/midi-qol#537]
In GitLab by @todlevi on Sep 22, 2021, 24:55

I'm using ItemMacro to check if a wolf hits then knocks the target prone. If the attack misses, no save is rolled, but failedSaves still contains the target. Is this by design? So I filter out the un-hit targets but that seems unnecessarily complicated.

```
// simpler code applies prone condition on a miss
// game.cub.addCondition("Prone", [...args[0].failedSaves]);

// I think this works 
let actually_failed = args[0].hitTargets.filter(t => args[0].failedSaves.includes(t));
game.cub.addCondition("Prone", [...actually_failed]);
```

**Comment by tposney** [comment]
First there is a bug in failed saves - which you report brought to my attention. Fix will go out in 0.8.57. 

Secondly, it's complicated, which is my way of saying it's not obvious to me what the correct values should be. It is complicated since people use varying degrees of automation in their game - and want to be abled test conditons when various rolls have not been made, i.e. attack hit is checked but not saves. If everything was full auto, failedSaves should only include those targets that were hit and did not save. In the case that saving throws are not being rolled failedSaves is not very well defined - and there are arguments that it should be all those that were hit etc.

I suspect that the right answer is if the attack missed the target is neither in saves or failed saves, which is what I'll push out in 0.8.57.

certainaly hitTargets filtered by failedSaves will work.

Probably the easiest way to do what you want is to use the new statusEffect active effect in dae, which avoids the duplicate icon problem and applies the current CUB definition for prone. (Assuming you are using DAE). This means you don't need a macro and the status effect will only be applied if they are hit and did not save.

**Comment by tposney** [comment]
In GitLab by @todlevi on Sep 23, 2021, 12:02

Hi, Tim. Thank you for your response and your excellent work on Midi QOL.
Core foundry should have more automation like yours.

You wrote: "Probably the easiest way to do what you want is to use the new
statusEffect active effect in dae, which avoids the duplicate icon problem
and applies the current CUB definition for prone. (Assuming you are using
DAE). This means you don't need a macro and the status effect will only be
applied if they are hit and did not save."

Can you briefly explain how I accomplish this or point me in the right
direction?

Thanks again!
Tod

<details><summary>...</summary>

On Wed, Sep 22, 2021 at 8:21 AM Tim Posney (@tposney) <gitlab@mg.gitlab.com> wrote:

> Issue was closed by Tim Posney
>
> —
> Reply to this email directly or view it on GitLab
> <https://gitlab.com/tposney/midi-qol/-/issues/537>.
> You're receiving this email because of your account on gitlab.com. If
> you'd like to receive fewer emails, you can unsubscribe
> <https://gitlab.com/-/sent_notifications/REDACTED/unsubscribe>
> from this thread or adjust your notification settings.
>

</details>
